helios-util/src/types/image_uri.rs

```diff
@@ -13,7 +14,7 @@ use thiserror::Error;
 #[error("invalid uri format, expected [domain.tld/]repo/image[:tag][@digest] format, got: {0}")]
 pub struct InvalidImageUriError(String);
 
-#[derive(Debug, Clone, Eq)]
+#[derive(State, Debug, Clone, Eq)]
```

Comment:
I think I'd prefer any Mahler state types were defined in helios-state (via newtypes if necessary) but it's fine for now.